src/App/Server/TrackerConnectionHandler.java

```diff
@@ -215,4 +216,30 @@ private String registerChatInStorage(String message) {
             return RegisterChatMessage.getNACK();
         }
     }
-}
\ No newline at end of file
+
+    private String searchMessage(String message) {
+        try {
+            GetChatMessage getChatMessage = GetChatMessage.fromString(message);
```

Comment:
this is incorrect, here we should have SearchChatMessage not history GetChatMessage